packages/core/src/utils/url-checker.ts

```diff
@@ -0,0 +1,73 @@
+import { getConfig } from '@/config';
+import { isValidUrl } from '@/utils';
+
+/**
+ * Check if a URL is the home URL.
+ *
+ * @param url The URL to check.
+ * @param ignoreProtocol - Ignore HTTP protocol when comparing URLs.
+ *
+ * @return Whether the URL is the home URL.
+ */
+export const isWPHomeUrl = (
+	url: string,
+	ignoreProtocol: boolean = true
+): boolean => {
+	if ( ! isValidUrl( url ) ) {
+		return false;
+	}
+	const { wpHomeUrl } = getConfig();
+
+	if ( ignoreProtocol ) {
+		return new URL( url ).host === new URL( wpHomeUrl ).host;
+	}
+
+	return url === wpHomeUrl || url.startsWith( wpHomeUrl );
+};
```

Comment:
Witespace + comments.

```suggestion
	if ( ! isValidUrl( url ) ) {
		return false;
	}

	const { wpHomeUrl } = getConfig();

	if ( ignoreProtocol ) {
		// Compare just the domains.
		return new URL( url ).host === new URL( wpHomeUrl ).host;
	}

	// Compare with the protocol.
	return url === wpHomeUrl || url.startsWith( wpHomeUrl );
};
```